File: backend/app/schemas/audit.py
from pydantic import BaseModel, ConfigDict

from app.schemas.types import LocalDatetime


class AuditLogRead(BaseModel):
    """Entrada de la bitácora de acciones."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int | None
    username: str | None
    role: str | None
    action: str
    module: str
    path: str
    entity_id: str | None
    status_code: int
    succeeded: bool
    ip_address: str | None
    created_at: LocalDatetime


class ModuleAccess(BaseModel):
    """Acceso de un perfil sobre un módulo del sistema."""

    code: str
    name: str
    can_view: bool
    can_manage: bool


class RolePermissions(BaseModel):
    role: str
    role_name: str
    modules: list[ModuleAccess]

File: frontend/src/types/audit.ts
export interface AuditLog {
  id: number
  user_id: number | null
  username: string | null
  role: string | null
  action: string
  module: string
  path: string
  entity_id: string | null
  status_code: number
  succeeded: boolean
  ip_address: string | null
  created_at: string
}

export interface ModuleAccess {
  code: string
  name: string
  can_view: boolean
  can_manage: boolean
}

export interface RolePermissions {
  role: string
  role_name: string
  modules: ModuleAccess[]
}

export interface AuditFilters {
  search?: string
  user_id?: number
  module?: string
  action?: string
  date_from?: string
  date_to?: string
  page: number
  page_size: number
}
